app.go

```diff
@@ -475,6 +506,39 @@ func New(opts ...Option) *App {
 		app.provide(p)
 	}
 
+	// If WithLogger and Printer are both provided, WithLogger takes precedence.
+	// If user supplied an fxlog constructor, we need to have dig's container
+	// initialize it and grab all of arguments from the graph and then return.
+	if app.fxlogConstructor != nil {
+		err := app.container.Provide(app.fxlogConstructor)
+		if err != nil {
+			err = fmt.Errorf("could not construct customer logger via fx.WithLogger: %w", err)
+			app.err = err
+
+			return app
+		}
+		var userLogger fxlog.Logger
+		if err := app.container.Invoke(func(got fxlog.Logger) { userLogger = got }); err != nil {
+			fxlog.Error("error constructing logger, did not get back fxlog.Logger").Write(app.logBuffer)
+			app.err = err
+
+			return app
```

Comment:
If app returns here, will anything written to app.logBuffer ever get flushed out of the buffer?

Reply:
This is valid. If the Invoke of the fxlog.Logger fails, that means we failed to construct the user provided logger.
In that case, we should as you're doing here, but then flush to the default logger and exit with an error.

Reply:
Yup, good call. Tests didn't catch this since I constructed a spy logger.